--- ./theano/gof/cmodule.py.orig	2014-03-12 20:00:00.000000000 -0600
+++ ./theano/gof/cmodule.py	2015-02-21 20:05:04.130272458 -0700
@@ -1723,7 +1723,7 @@ class GCC_compiler(object):
         # or 64 bit and compile accordingly. This step is ignored for ARM
         # architectures in order to make Theano compatible with the Raspberry
         # Pi.
-        if any([not 'arm' in flag for flag in cxxflags]):
+        if all([not 'arm' in flag for flag in cxxflags]):
             n_bits = local_bitwidth()
             cxxflags.append('-m%d' % n_bits)
             _logger.debug("Compiling for %s bit architecture", n_bits)
