--- ./doc/tutorial/using_gpu.txt.orig	2015-03-26 15:34:26.000000000 -0600
+++ ./doc/tutorial/using_gpu.txt	2015-03-31 20:00:00.000000000 -0600
@@ -670,7 +670,7 @@ Modify and execute to work for a matrix
 
 
 
-.. _pyCUDA_theano:
+.. _pyCUDA_theano_example:
 
 **Example: Theano + PyCUDA**
 
--- ./MANIFEST.in.orig	2015-03-26 15:34:24.000000000 -0600
+++ ./MANIFEST.in	2015-03-31 20:00:00.000000000 -0600
@@ -5,7 +5,7 @@ global-include *.cuh
 global-include *.h
 global-include *.sh
 global-include *.pkl
-recursive-include docs
+recursive-include docs *
 include bin/theano-cache
 include bin/theano-nose
 include bin/theano-test
--- ./theano/sandbox/cuda/basic_ops.py.orig	2015-03-26 15:34:27.000000000 -0600
+++ ./theano/sandbox/cuda/basic_ops.py	2015-03-31 20:00:00.000000000 -0600
@@ -1239,8 +1239,10 @@ class GpuCAReduce(GpuOp):
 
     def c_code_reduce_01X(self, sio, node, name, x, z, fail, N):
         """
-        :param N: the number of 1 in the pattern N=1 -> 01, N=2 -> 011 N=3 ->0111
-                  Work for N=1,2,3
+        :param N: the number of 1s in the pattern N=1 -> 01, N=2 -> 011,
+            N=3 ->0111.  Works for N=1,2,3
+
+        Reduce.
         """
 
         assert N in [1, 2, 3]
